### PR TITLE
Use nose and unittest modules for pws tests

### PR DESCRIPTION
**Pull request by cco3**
Additionally, this change comments out the url shortening test
since that was already broken.

@mmocny

### CODE REVIEW

File: web-service/tests.py [hunk 1]
```diff
@@ -16,153 +16,197 @@
...
+
+class PwsTest(unittest.TestCase):
+    _HOST = None  # Set in main()
+
+    @property
+    def HOST(self):
+        PwsTest._HOST
+
+    def request(self, params=None, payload=None):
+        PATH = getattr(self, 'PATH')
```
Comment (mmocny) [inline web-service/tests.py]: Why not self.PATH?

Comment (cco3) [inline web-service/tests.py]: Originally I didn't always have it defined, so there was a default value.  I'll change it.

File: web-service/tests.py [hunk 2]
```diff
@@ -16,153 +16,197 @@
...
+        JSON = getattr(self, 'JSON', False)
+        url = '{}/{}'.format(self.HOST, PATH)
+        if params:
+            url += '?{}'.format(urllib.urlencode(params))
+        args = [url]
+        if payload is not None:
+            args.append(json.dumps(payload))
+        req = urllib2.Request(*args)
```
Comment (mmocny) [inline web-service/tests.py]: May you add a line to tell that this changes from POST/GET depending on args.

**Comment by mmocny** [comment]
I'm on the run, so haven't run this.  But it looks quite nice, a big improvement.

Curious if the UrlShortener *test* isn't working, or the url shortening on the server is failing?

**Comment by cco3** [comment]
Dunno...it didn't work when I first ran the tests so I didn't really mess with it.  It bombs out when trying to de-serialize the json.  My guess is that you are returning a bare url without quotes or anything.  Let me see if it's that quick of a fix.

**Comment by mmocny** [comment]
Ah yes, we pass through exactly what goo.gl gives.

**Comment by cco3** [comment]
The server returns a 400, so it's not that simple.

**Comment by mmocny** [comment]
Is it possibly that you don't have a config.SECRET.json file?  goo.gl
shortener requires a valid api key.. try running the test against the
deployed url-caster-dev.

**Comment by cco3** [comment]
Ah...that would explain it...yuck...

**Comment by cco3** [comment]
Ok, done.

**Comment by mmocny** [comment]
This is really quite awesome, amazing work.

I keep getting a test failure, but landing this since I think thats a separate issue.

**Comment by mmocny** [comment]
May you also add a document describing how to run tests (including `pip install nose`) some time?

**Pull request merged**